$OpenBSD: patch-muttlib_c,v 1.4 2013/03/14 10:14:10 sthen Exp $

fix segfault when $message_cachedir is set and opening a POP3 mailbox.
upstream 1a4c43138685, fixes bug#3457

--- muttlib.c.orig	Wed Mar 13 16:22:39 2013
+++ muttlib.c	Wed Mar 13 16:22:40 2013
@@ -1962,6 +1962,7 @@ void mutt_encode_path (char *dest, size_t dlen, const 
 {
   char *p = safe_strdup (src);
   int rc = mutt_convert_string (&p, Charset, "utf-8", 0);
-  strfcpy (dest, rc == 0 ? p : src, dlen);
+  /* `src' may be NULL, such as when called from the pop3 driver. */
+  strfcpy (dest, (rc == 0) ? NONULL(p) : NONULL(src), dlen);
   FREE (&p);
 }
